server/src/grpc/meta_event_service/mod.rs

```diff
@@ -317,21 +360,25 @@ async fn handle_open_shard(ctx: HandlerContext, request: OpenShardRequest) -> Re
 }
 
 async fn do_close_shard(ctx: &HandlerContext, shard_id: ShardId) -> Result<()> {
-    let tables_of_shard =
-        ctx.cluster
-            .freeze_shard(shard_id)
-            .await
-            .box_err()
-            .context(ErrWithCause {
-                code: StatusCode::Internal,
-                msg: "fail to freeze shard before close it in cluster",
-            })?;
+    let shard = ctx
+        .cluster
+        .get_shard(shard_id)
+        .with_context(|| ErrNoCause {
+            code: StatusCode::Internal,
+            msg: format!("shard not found when closing shard, shard_id:{shard_id}",),
+        })?;
+
+    // Lock for serializing the write operation.
+    let _guard = shard.serializing_lock.lock().await;
+    info!("Shard close sequentially begin, shard_id:{shard_id}");
+
+    shard.data.freeze();
```

Comment:
If the `serial_lock` is introduced, I guess the freeze is not necessary anymore.